src/main/java/org/jabref/gui/actions/ExportLinkedFilesAction.java

```diff
@@ -0,0 +1,169 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import javax.swing.AbstractAction;
+
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+
+import org.jabref.Globals;
+import org.jabref.JabRefGUI;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.FXDialogService;
+import org.jabref.gui.util.DefaultTaskExecutor;
+import org.jabref.gui.util.DirectoryDialogConfiguration;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.OS;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.util.FileHelper;
+import org.jabref.model.util.OptionalUtil;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.controlsfx.dialog.ProgressDialog;
+
+public class ExportLinkedFilesAction extends AbstractAction {
+
+    private static final Log LOGGER = LogFactory.getLog(ExportLinkedFilesAction.class);
+    private final BiFunction<Path, Path, Path> resolvePathFilename = (path, file) -> {
+        return path.resolve(file.getFileName());
+    };
+    private final DialogService ds = new FXDialogService();
+    private long totalFilesCount;
+    private BibDatabaseContext databaseContext;
+    private Optional<Path> exportPath = Optional.empty();
+    private List<BibEntry> entries;
+
+    public ExportLinkedFilesAction() {
+        super(Localization.lang("Copy attached files to folder..."));
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
+        DirectoryDialogConfiguration dirDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(Paths.get(Globals.prefs.get(JabRefPreferences.EXPORT_WORKING_DIRECTORY)))
+                .build();
+        entries = JabRefGUI.getMainFrame().getCurrentBasePanel().getSelectedEntries();
+        exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> ds.showDirectorySelectionDialog(dirDialogConfiguration));
+
+        exportPath.ifPresent(path -> {
+            databaseContext = JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext();
+            totalFilesCount = entries.stream().flatMap(entry -> entry.getFiles().stream()).count();
+
+            Service<Void> exportService = new ExportService();
+            startServiceAndshowProgessDialog(exportService);
+        });
+    }
+
+    private <V> void startServiceAndshowProgessDialog(Service<V> service) {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            service.start();
+
+            ProgressDialog progressDialog = new ProgressDialog(service);
```

Comment:
Can you please extract the "show progress dialog" part of this method to the `DialogService` class.